File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map <char, int> m ;
        int left = 0;
        int result = 0;
        for (int right=0; right<s.size(); right++)
        {
            m[s[right]]++;
            while( m[s[right]] > 1)
            {
                m[s[left]]--;
                left +=1;
        }
        result = max(result , (right-left) + 1);
        }
        return result;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
       HashMap<Character,Integer> hash=new HashMap<>();
        int start=0;
        int maxLen=0;
        for(int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
            if (hash.containsKey(c) && hash.get(c) >= start) 
                start = hash.get(c) + 1;
            hash.put(c, i);
            maxLen = Math.max(maxLen, i - start + 1);
        }
        return maxLen;
    }
}